File: client/lib/features/pos/domain/payment_method.dart
/// Sent/received as a plain integer — see onboarding_enums.dart for why.
/// Mirrors Purch.Domain.Enums.PaymentMethod exactly, in the same declared
/// order. Only cash, bankTransfer, and manualGcashQr are wired to a working
/// checkout flow so far — qrPh needs a live Xendit webhook, billPaymentELoad
/// needs the Dragonpay integration from ADR 0004, utangCredit is Phase 9 per
/// the implementation plan, and split (multi-method) isn't built yet.
enum PaymentMethod {
  cash,
  qrPh,
  bankTransfer,
  manualGcashQr,
  billPaymentELoad,
  utangCredit,
  split,
}

/// Mirrors Purch.Domain.Enums.PaymentStatus exactly, in the same declared order.
enum PaymentStatus { pending, confirmed, failed, cancelled }

File: backend/src/Purch.Domain/Enums/PaymentMethod.cs
namespace Purch.Domain.Enums;

/// <summary>
/// No Card, no per-wallet methods (GCash/Maya) as gateway-processed payments —
/// digital gateway payment is consolidated behind QR Ph (one PSP-agnostic
/// scan). ManualGcashQr is not a gateway integration: it's a merchant-
/// uploaded static QR Ph code paid straight into the tenant's own account,
/// confirmed by the cashier the same way BankTransfer is. See docs/adr and
/// the implementation plan.
/// </summary>
public enum PaymentMethod
{
    Cash,
    QrPh,
    BankTransfer,
    ManualGcashQr,
    BillPaymentELoad,
    UtangCredit,
    Split,
}
